python/hsfs/core/feature_monitoring_config.py

```diff
@@ -479,7 +483,7 @@ def run_job(self):
         # Returns
             `Job`. A handle for the job computing the statistics.
         """
-        if not self._id:
+        if not self._id or not self._job_name:
```

Comment:
Could it be that the FM config is created but doesn't have a job attached to it?
If so, we should update the exception message accordingly.

Reply:
In principle no, there should not be a case where it is registered and there is no job name. The message is not specific to id or job name in that regard so I think it is fine.